### PR TITLE
Update the documentation of VectorTools::project.

### PR DESCRIPTION
**Pull request by bangerth**

### CODE REVIEW

**Comment by tjhei** [comment]
This somewhat overlaps with the description in the namespace. Would it make sense to  merge and remove one or the other and reference the other location?

**Comment by bangerth** [comment]
On 1/10/19 3:18 PM, Timo Heister wrote:
> This somewhat overlaps with the description in the namespace. Would it make 
> sense to merge and remove one or the other and reference the other location?

There is already a reference to namespace in the description of the function 
and the other way around.

I've long wanted to remove the descriptions of individual functions in the 
namespace documentation because that's really not where it ought to be. But I 
don't have the time for it right now and thought this is strictly progress...

**Comment by tjhei** [comment]
> I've long wanted to remove the descriptions of individual functions in the namespace documentation because that's really not where it ought to be.

It is harder to find, but the advantage is that we can have one place to document all variants of ``project`` instead of having to say "go see one of the other versions of project() that I can not link to here". Either way, I think there is value in having only one location where things are explained.


> . But I don't have the time for it right now and thought this is strictly progress...

Except that one day somebody has to read through both and unify it. :-)

File: include/deal.II/numerics/vector_tools.h [hunk 1]
```diff
@@ -798,12 +798,35 @@ namespace VectorTools
...
+   * $M_{ij} = (\varphi_i, \varphi_j)_\Omega$ and
+   * $\Phi_i = (\varphi_i,f)_\Omega$.
+   * By default, no boundary values for $f_h$ are
+   * imposed, but there are optional parameters to this function that allow
+   * imposing either zero boundary values or, in a first step, to project
+   * the boundary values of $f$ onto the finite element space on the boundary
+   * of the mesh in a similar way to above, and then using these values as the
+   * imposed boundary values for $f_h$.
+   *
+   * By default, this projection to the boundary or enforcement of zero
```
Comment (tjhei) [inline include/deal.II/numerics/vector_tools.h]: you are repeating yourself (paragraph above)

File: include/deal.II/numerics/vector_tools.h [hunk 2]
```diff
@@ -798,12 +798,35 @@ namespace VectorTools
...
+   * boundary values are disabled. The ordering of arguments to this function
+   * is such that you need not give a second quadrature formula if you don't
+   * want to project to the boundary first, but that you must if you want to
+   * do so.
```
Comment (tjhei) [inline include/deal.II/numerics/vector_tools.h]: maybe say that what the second quadrature does? reference argument name?

**Comment by bangerth** [comment]
I've addressed @tjhei 's comments.

**Comment by masterleinad** [comment]
This is just documentation so we don't need the tester here.

**Pull request merged**